### PR TITLE
Code Quality: Misc fixes and bumping dependencies

### PR DESCRIPTION
**Pull request by hez2010**
Although we cannot enable trimming for now.

### CODE REVIEW

**Comment by hez2010** [comment]
Ready for review

File: src/Files.OpenDialog/FilesLauncher/FilesLauncher.vcxproj [hunk 1]
```diff
@@ -381,12 +381,12 @@
...
+    <Import Project="..\..\..\packages\Microsoft.Windows.ImplementationLibrary.1.0.220914.1\build\native\Microsoft.Windows.ImplementationLibrary.targets" Condition="Exists('..\..\..\packages\Microsoft.Windows.ImplementationLibrary.1.0.220914.1\build\native\Microsoft.Windows.ImplementationLibrary.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
-      <ErrorText>Questo progetto fa riferimento a uno o più pacchetti NuGet che non sono presenti in questo computer. Usare lo strumento di ripristino dei pacchetti NuGet per scaricarli. Per altre informazioni, vedere http://go.microsoft.com/fwlink/?LinkID=322105. Il file mancante è {0}.</ErrorText>
+      <ErrorText>このプロジェクトは、このコンピューター上にない NuGet パッケージを参照しています。それらのパッケージをダウンロードするには、[NuGet パッケージの復元] を使用します。詳細については、http://go.microsoft.com/fwlink/?LinkID=322105 を参照してください。見つからないファイルは {0} です。</ErrorText>
```
Comment (QuaintMako) [inline src/Files.OpenDialog/FilesLauncher/FilesLauncher.vcxproj:388]: Maybe the text error should be in English?

File: src/Files.Shared/Extensions/LinqExtensions.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ public static class LinqExtensions
 			return defaultValue;
 		}
 
-		public static Task<TValue?> GetAsync<TKey, TValue>(this IDictionary<TKey, Task<TValue?>> dictionary, TKey key, Func<Task<TValue?>> defaultValueFunc)
+		public static Task<TValue?> GetAsync<TKey, TValue>(this IDictionary<TKey, Task<TValue?>> dictionary, TKey key, Func<Task<TValue?>> defaultValueFunc) where TKey : notnull
 		{
 			if (dictionary is null || key is null)
```
Comment (QuaintMako) [inline src/Files.Shared/Extensions/LinqExtensions.cs:46]: Does the `key is null` in the condition line 46 still makes sense with the `where TKey : notnull` ?

File: builds/azure-pipelines.yml [hunk 3]
```diff
@@ -62,7 +62,7 @@ steps:
 - task: UseDotNet@2
   inputs:
     packageType: sdk
-    version: 6.0.400
+    version: 7.0.0
```
Comment (yaichenbaum) [inline builds/azure-pipelines.yml:65]: @hez2010 do you know which version we need?

Comment (hez2010) [reply builds/azure-pipelines.yml]: Oops I forgot to update. We can simply use `7.0.*` and set `include-prerelease: true`.

Comment (hez2010) [reply builds/azure-pipelines.yml]: # Resolved.

File: src/Files.Shared/Extensions/LinqExtensions.cs (hunk 2 above)
Comment (hez2010) [reply src/Files.Shared/Extensions/LinqExtensions.cs:46]: `where TKey : notnull` doesn't enforce non-nullable unless we treat all nullable related warnings as errors.

File: builds/azure-pipelines.yml (hunk 3 above)
Comment (yaichenbaum) [reply builds/azure-pipelines.yml]: Still failing :thinking:
We might need to specify the exact version number

Comment (hez2010) [reply builds/azure-pipelines.yml]: The exact sdk version number is `7.0.100-rc.2.22477.23`.

Comment (hez2010) [reply builds/azure-pipelines.yml]: Why MSBuild cannot find the installed sdk?

Comment (hez2010) [reply builds/azure-pipelines.yml]: Now things go right. 
We need to write a `UsePreviews=True` to sdk.txt in VS user data directory to enable preview SDK adoption for VS.

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**